woocommerce/woocommerce-ios
# Blaze: Add error screen for campaign creation failure

Closes: #11724 


## Description

This PR adds a new error modal for campaign creation failures. It also handles the dismissal of the campaign creation flow using the new coordinator.

Extra: Renamed the CTA on the Blaze section on the dashboard screen from "Create a campaign" to "Promote".

## Testing instructions

- Update `BlazeConfirmPaymentViewModel`'s method `submitCampaign` to force `shouldDisplayCampaignCreationError` to be true and build the app.
- Log in to a store that's eligible for Blaze.
- If the campaign creation entry point is available on the Dashboard screen, select it and select a product if needed. - Otherwise, navigate to the Products tab, select a published product and tap Promote with Blaze.
- On the campaign creation form, tap Confirm Details.
- On the payment screen, tap Submit Campaign.
- Notice that an error screen is displayed. Confirm the following:
  - The UI is correct.
  - Tap on Get support, the support form should be displayed.
  - Tap on Retry should dismiss the view, but currently it doesn't because we are forcing the error state.
  - Tap on Cancel should dismiss the whole creation flow. You can test when the creation form is presented and when it's pushed to the existing navigation stack.

Feel free to disable the feature flag to test that the web view is dismissed when campaign creation succeeds.


## Screenshots


https://github.com/woocommerce/woocommerce-ios/assets/5533851/d055aaaa-ddbe-41ad-a939-56ee703b11f0



---
